### PR TITLE
Test/pipeline converstion

### PR DESCRIPTION
**Pull request by alexvy86**
## Description

Testing move of the `Build - client packages` pipeline to new base template.

## Reviewer Guidance

The review process is outlined on [this wiki page](https://github.com/microsoft/FluidFramework/wiki/PR-Guidelines#guidelines).

[AB#5816](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/5816)

### CODE REVIEW

File: tools/pipelines/templates/include-generate-notice-steps.yml
Comment (alexvy86) [inline tools/pipelines/templates/include-generate-notice-steps.yml:1]: I'd say revert this file... all the tool did in this file is remove the trailing newline, which we should keep.

Same for `include-git-tag-steps.yml` and `include-install-pnpm.yml` (there it also removed comments, so more reason for reverting).

File: tools/pipelines/templates/include-policy-check.yml [hunk 1]
```diff
@@ -28,14 +28,14 @@ stages:
...
   dependsOn: [] # Has no prereqs
   displayName: Policy checks
-  pool: Small
+  pool: Small-1ES
   jobs:
   - job:
     displayName: Run checks
     steps:
-    - template: include-use-node-version.yml
+    - template: /tools/pipelines/templates/1ES/include-use-node-version.yml@self
```
Comment (alexvy86) [inline tools/pipelines/templates/include-policy-check.yml:36]: Not sure the non-1ES templates should include the 1ES ones or run in the 1ES pools... unless this is already preparation to move everything to the new templates, but in that case maybe let's just update all the templates inline at their current location?

File: tools/pipelines/templates/include-install.yml [hunk 2]
```diff
@@ -25,4 +25,4 @@ steps:
       targetType: 'inline'
       workingDirectory: ${{ parameters.buildDirectory }}
       script: |
-        ${{ parameters.packageManagerInstallCommand }}
\ No newline at end of file
+        ${{ parameters.packageManagerInstallCommand }}
```
Comment (alexvy86) [inline tools/pipelines/templates/include-install.yml:28]: Restore trailing newline

**Review by alexvy86 — COMMENTED** [review]
Left a few comments; I have more similar ones but this is still draft so might be early to comment on everything.

We've already talked about restoring comments and newlines removed by the tool... I'd also suggest we restore the trailing newlines it removed from the last line in some files.

File: tools/pipelines/templates/include-install-pnpm.yml [hunk 3]
```diff
@@ -39,13 +39,10 @@ steps:
   inputs:
     targetType: 'inline'
     workingDirectory: ${{ parameters.buildDirectory }}
-    # The previous task (cache restoration) can timeout, which is classified as canceled, but since it's just cache
```
Comment (rohandubal) [inline tools/pipelines/templates/include-install-pnpm.yml:42]: Need to bring this comment in.

File: tools/pipelines/templates/include-install-pnpm.yml [hunk 4]
```diff
@@ -39,13 +39,10 @@ steps:
...
-    # restoration, we want to continue even if it timed out.
     condition: or(succeeded(), canceled())
-    # workspace-concurrency 0 means use use the CPU core count. This is better than the default (4) for larger agents.
```
Comment (rohandubal) [inline tools/pipelines/templates/include-install-pnpm.yml:45]: Same here, need to bring this comment in.

File: tools/pipelines/templates/include-use-node-version.yml [hunk 5]
```diff
@@ -1,14 +1,9 @@
-# Copyright (c) Microsoft Corporation and contributors. All rights reserved.
-# Licensed under the MIT License.
-
-# include-use-node-version template to use a specific node version
-
```
Comment (rohandubal) [inline tools/pipelines/templates/include-use-node-version.yml:5]: Missed reverts, need to get back comments and license header

File: tools/pipelines/templates/include-use-node-version.yml [hunk 6]
```diff
@@ -1,14 +1,9 @@
...
+# This Yaml Document has been converted by ESAI Yaml Pipeline Conversion Tool.
+# Please make sure to check all the converted content, it is your team's responsibility to make sure that the pipeline is still valid and functions as expected.
 parameters:
-# The node version required. The version must already be installed in the image.
```
Comment (rohandubal) [inline tools/pipelines/templates/include-use-node-version.yml:7]: Another one

File: tools/pipelines/templates/1ES/build-npm-package.yml [hunk 7]
```diff
@@ -426,11 +429,12 @@ extends:
                   # At this point we want to publish the artifact with the bundle size analysis,
                   # but as part of 1ES migration that's now part of templateContext.outputs below. TODO: Delete/Update this comment on successful inline test
                   - task: 1ES.PublishPipelineArtifact@1
-                      displayName: Publish Artifacts - bundle-analysis
-                      condition: and( succeeded(), ne(variables['Build.Reason'], 'PullRequest'), eq(${{ parameters.taskPublishBundleSizeArtifacts }}, true) )
-                      targetPath: ${{ parameters.buildDirectory }}/artifacts/bundleAnalysis
-                      artifactname: bundleAnalysis
-                      publishLocation: Container
+                  inputs:
```
Comment (alexvy86) [inline tools/pipelines/templates/1ES/build-npm-package.yml:432]: Indentation here is funky. Also, displayName and condition should be at the same level as inputs, IIRC; the rest are children of inputs.

File: tools/pipelines/templates/1ES/build-docker-service.yml [hunk 8]
```diff
@@ -415,7 +415,7 @@ extends:
           - task: 1ES.PushContainerImage@1
             displayName: Docker Push
             inputs:
-              image: ${{ variables.buildContainerName }}:$(containerTagSuffix)
+              image: $(containerRegistryUrl)/${{ variables.buildContainerName }}:$(containerTagSuffix)
```
Comment (alexvy86) [inline tools/pipelines/templates/1ES/build-docker-service.yml:418]: I think `buildContainerName` already has a "prefix" that acts as registry. See how we define it [here](https://github.com/microsoft/FluidFramework/blob/a3dfbb6ac3c7c8893734907eaeef13d03b149d2b/tools/pipelines/templates/build-docker-service.yml#L108-L113).

Comment (seanimam) [reply tools/pipelines/templates/1ES/build-docker-service.yml]: Updated.

File: tools/pipelines/templates/1ES/build-docker-service.yml [hunk 9]
```diff
@@ -403,20 +410,20 @@ extends:
 
         # Build
         - task: 1ES.BuildContainerImage@1
-          displayName: Docker Build - ${{ variables.buildContainerName }}
+          displayName: Docker Build - $(buildContainerName)
```
Comment (alexvy86) [inline tools/pipelines/templates/1ES/build-docker-service.yml:413]: Let's see what happens here, not sure how this will act on a display name.

**Review by alexvy86 — COMMENTED** [review]
The changes you made look good... but I think there's still some lines that we'll need to change. E.g. line 143 (and a few other instances right after this) is still trying to use variables at compile time:

```
        - ${{ if eq(variables.testBuild, true) }}:
          - name: buildContainerName
            value: test/${{ parameters.containerName }}
        - ${{ if eq(variables.testBuild, false) }}:
          - name: buildContainerName
            value: build/${{ parameters.containerName }}
```

Or line 272:

`- ${{ if eq(variables.componentDetection, true) }}:`

Line 276 does have a fix, I believe. Instead of 

`requireNotice: ${{ eq(variables.releaseImage, true) }}`

Use

`requireNotice: $[eq(variables.releaseImage, true)]`

**Comment by alexvy86** [comment]
This work has been completed in other PRs, closing initial draft.

**Pull request closed without merging**